src/settings.js

```diff
@@ -108,6 +108,12 @@ var TOTAL_STACK = 5*1024*1024;
 // What malloc()/free() to use, out of
 //  * dlmalloc - a powerful general-purpose malloc
 //  * emmalloc - a simple and compact malloc designed for emscripten
+//  * emmalloc-debug - use emmalloc and add extra assertion checks
+//  * emmalloc-memvalidate - use emmalloc with assertions+heap consistency
+//                           checking.
+//  * emmalloc-verbose - use emmalloc with assertions + verbose logging.
+//  * emmalloc-memvalidate-verbose - use emmalloc with assertions + heap
+//                                   consistency checking + verbose logging.
```

Comment:
I'm in the process of designing some way to opt into building and using debug system libraries across the board.  I originally was going with https://github.com/emscripten-core/emscripten/pull/13078.. but I'm going to abandon that in favor a separate opt-in debug libraries.

Perhaps that will enable use to eliminate some of these configurations .. ?   But it doesn't seem to bad to add them here and then remove them once there is a more general mechanism.

Reply:
`emmalloc-debug`  is the mode that builds when `ASSERTIONS=2` are enabled. But emmalloc-memvalidate and emmalloc-verbose options are both really heavy-weight and super-slow for a specific memory allocator debugging/introspection work, so that would completely break anyone using `ASSERTIONS=2` for any other debugging purposes. That is why to me it did not make sense to lump emmalloc-memvalidate or emmalloc-verbose to existing debug layer.

Reply:
Sounds reasonable to me.